File: frontend/src/types/preset.ts
export type PresetCategory = 'general' | 'debug' | 'testing' | 'refactor' | 'docs' | 'review' | 'custom'
export type PresetScope = 'global' | 'project'

export type Preset = {
  id: string
  name: string
  description?: string | null
  command: string
  category?: PresetCategory
  shortcut?: string | null
  created_at?: string | null
  updated_at?: string | null
}

export type PresetsResponse = {
  global: Preset[]
  project: Preset[]
}

File: backend/app/models/preset.py
from __future__ import annotations

from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, Field


class PresetVariable(BaseModel):
    name: str
    description: str
    default: Optional[str] = None


class PresetCategory(str, Enum):
    GENERAL = "general"
    DEBUG = "debug"
    TESTING = "testing"
    REFACTOR = "refactor"
    DOCS = "docs"
    REVIEW = "review"
    CUSTOM = "custom"


class PresetScope(str, Enum):
    GLOBAL = "global"
    PROJECT = "project"


class Preset(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    command: str
    variables: List[PresetVariable] = Field(default_factory=list)
    category: PresetCategory = PresetCategory.GENERAL
    shortcut: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class PresetCreate(BaseModel):
    name: str
    description: Optional[str] = None
    command: str
    category: PresetCategory = PresetCategory.GENERAL
    shortcut: Optional[str] = None
    scope: PresetScope
    project_path: Optional[str] = None


class PresetUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    command: Optional[str] = None
    category: Optional[PresetCategory] = None
    shortcut: Optional[str] = None
